pkgs/by-name/cp/cppadcodegen/package.nix

```diff
@@ -0,0 +1,195 @@
+{
+  adolc,
+  cmake,
+  cppad,
+  doxygen,
+  eigen,
+  fetchFromGitHub,
+  graphviz,
+  gtest,
+  lib,
+  llvmPackages,
+  makeFontsConf,
+  pkg-config,
+  stdenv,
+  texlive,
+  valgrind,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cppadcodegen";
+  version = "2.5.0";
+
+  src = fetchFromGitHub {
+    owner = "joaoleal";
+    repo = "CppADCodeGen";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-na8o+bqzign2nSk5AQdkIVQm3CIb0oFqEneGnYKQDyg=";
+  };
+
+  patches = [
+    # Use gtest from nixpkgs instead of downloading it from github
+    # ref https://github.com/joaoleal/CppADCodeGen/pull/90
+    ./90.patch
```

Comment:
Shouldn't it be better to give a descriptive name for the patch (maybe the pr name)?

Reply:
I must admit I just `wget https://github.com/joaoleal/CppADCodeGen/pull/90.patch`.
I am open to suggestions. Name is a good one.

Reply:
i'm thinking that it may be better to directly fetch the patch using `fetchpatch2`?

Reply:
As long as it is not merged, the hash of that url might change, because of git pushs that I could have to make to please upstream author. And the commit could also disappear in the case of git push -f

Reply:
Now using "{pr_number}_{pr_title|slug}.patch"